docs/docs/api-ref.md

```diff
@@ -5,257 +5,3010 @@ layout: docs
+
+
+**Returns**
 
-Populates a list of views for a workbook object. You must populate views before you can iterate through the views.
+An updated `DatasourceItem`.
+
+
+**Example**
 
 ```py
-Workbooks.populate_views(workbook_obj)
+# from server-client-python/test/test_datasource.py
+
+    ...
+
+    single_datasource = TSC.DatasourceItem('test', '1d0304cd-3796-429f-b815-7258370b9b74')
+    # need to specify the ID - this is a workaround. 
+    single_datasource._id = '9dbd2263-16b5-46e1-9c43-a76bb8ab65fb'
+    single_datasource._tags = ['a', 'b', 'c']
+    single_datasource._project_name = 'Tester'
+    updated_datasource = self.server.datasources.update(single_datasource)
+
+     ...
+
 ```
 
-### Populate Connections for a Workbook
 
-Populates a list of connections for a given workbook. You must populate connections before you can iterate through the
-connections.
 
-```py
-Workbooks.populate_connections(workbook_obj)
+<br>   
+<br>  
+
+## Filters
+
+The TSC library provides a `Filter` class that you can use to filter results returned from the server. 
+
+You can use the `Filter` and `RequestOptions` classes to filter and sort the following endpoints:
+
+- Users
+- Datasources
+- Workbooks
+- Views 
+
+For more information, see [Filter and Sort](filter-sort). 
+
+
+### Filter class
+
+```py  
+Filter(field, operator, value)
 ```
 
-### Populate a Preview Image for a Workbook
+The `Filter` class corresponds to the *filter expresions* in the Tableau REST API. 
 
-Populates a preview image for a given workbook. You must populate the image before you can iterate through the
-connections.
+
+
+**Attributes**
+
+Name | Description  
+:--- | :--- 
+`Field` | Defined in the `RequestOptions.Field` class.
+`Operator` | Defined in the `RequestOptions.Operator` class
+`Value` | The value to compare with the specified field and operator.
+
+
+
+
+
+<br>  
+<br>   
+
+
+## Groups
+
+Using the TSC library, you can get information about all the groups on a site, you can add or remove groups, or add or remove users in a group.
+
+The group resources for Tableau Server are defined in the `GroupItem` class. The class corresponds to the group resources you can access using the Tableau Server REST API. The group methods are based upon the endpoints for groups in the REST API and operate on the `GroupItem` class.
+
+<br>   
+<br> 
+
+### GroupItem class
 
 ```py
-Workbooks.populate_connections(workbook_obj)
+GroupItem(name)
 ```
 
-### Get Views for a Workbook
+The `GroupItem` class contains the members or attributes for the view resources on Tableau Server. The `GroupItem` class defines the information you can request or query from Tableau Server. The class members correspond to the attributes of a server request or response payload.
```

Comment:
yep. Fixed that. Also changed "view resources" to "group resources"  - copy and paste error.